pkg/controllermanager/controller/shoot/shoot.go

```diff
@@ -143,6 +147,12 @@ func NewShootController(
 		UpdateFunc: shootController.shootReferenceUpdate,
 	})
 
+	shootInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    shootController.shootRetryAdd,
+		UpdateFunc: shootController.shootRetryUpdate,
+		// TODO: check that we handle also the deletion case
```

Comment:
WDYM - what needs to be handled in this case?

Reply:
Nice catch, it was a leftover. I removed it now.